src/crimson/osd/pg.cc

```diff
@@ -661,7 +661,24 @@ seastar::future<Ref<MOSDOpReply>> PG::handle_failed_op(
   }, load_obc_ertr::assert_all{ "can't live with object state messed up" });
 }
 
-seastar::future<Ref<MOSDOpReply>> PG::do_osd_ops(
+crimson::errorator<crimson::ct_error::eagain>::future<> PG::rep_repair_primary_object(
+  Ref<MOSDOp> m,
+  const hobject_t& oid,
+  eversion_t& v) 
+{
+  ceph_assert(is_primary());
+  logger().debug("{}: {} peers osd.{}", __func__, oid, get_acting_recovery_backfill());
+  // Add object to PG's missing set if it isn't there already
+  if (!get_local_missing().is_missing(oid)) {
```

Comment:
if the object is missing, `is_unreadable_object(oid)` should return `true`, so we should have handle this case and start an urgent recovery right in `ClientRequest::process_op()` in the "recover_missing" stage.

Reply:
Is it not possible that at the stage of checking `is_unreadable_object(oid)` the object was readable but then by the time we called the `pg_backend` functions the object became unreadable?